plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -200,17 +200,25 @@ const transformProps = (chartProps: TableChartProps): TableChartTransformedProps
     server_page_length: serverPageLength = 10,
     order_desc: sortDesc = false,
     query_mode: queryMode,
+    show_totals: showTotals,
   } = formData;
 
   const [metrics, percentMetrics, columns] = processColumns(chartProps);
   const data = processDataRecords(queriesData?.[0]?.data, columns);
-  const rowCount = queriesData?.[1]?.data?.[0]?.rowcount as number;
+  const rowCount = serverPagination
+    ? (queriesData?.[1]?.data?.[0]?.rowcount as number)
+    : queriesData?.[0]?.rowcount;
+  const totals =
+    showTotals &&
+    queryMode === QueryMode.aggregate &&
+    queriesData?.[queriesData.length - 1]?.data[0];
```

Comment:
To make finding the correct query easier, I think we should add an optional `id` field to `QueryObject` that gets passed to the result. I can open a PR for this.